src/saveEventsFile.m

```diff
@@ -289,32 +322,71 @@ function printHeaderExtraColumns(logFile)
 
         logFile = checklLogFile('fields', logFile, iEvent, cfg);
 
-        onset = logFile(iEvent).onset;
-        duration = logFile(iEvent).duration;
-        trial_type = logFile(iEvent).trial_type;
-
-        % we skip events with onset or duration that are empty, nan or char
-        if any(cell2mat(cellfun(@isnan, {onset duration}, 'UniformOutput', false))) || ...
-                any(cellfun(@ischar, {onset duration})) || ...
-                any(isempty({onset duration}))
-
-            warning('saveEventsFile:emptyEvent', ...
-                    '\nSkipping saving this event.\n onset: %s \n duration: %s\n', ...
-                    onset, ...
-                    duration);
-
+        % check if this event should be skipped
+        skipEvent = false; 
+        
+        % if this is _events file, we skip events with onset or duration 
+        % that are empty, nan or char. 
+        if logFile(1).isStim==0 
+            
+            onset = logFile(iEvent).onset;
+            duration = logFile(iEvent).duration;
+            trial_type = logFile(iEvent).trial_type;
+
+            if any(cell2mat(cellfun(@isnan, {onset duration}, 'UniformOutput', false))) || ...
+               any(cellfun(@ischar, {onset duration})) || ...
+               any(isempty({onset duration})) 
+
+                skipEvent = true; 
+
+                warningMessage = sprintf(['saveEventsFile:emptyEvent', ...
+                        '\nSkipping saving this event.\n onset: %s \n duration: %s\n'], ...
+                        onset, ...
+                        duration);
+            end
+            
+        % if this is _stim file, we skip missing events (i.e. events where 
+        % all extra columns have NO values)
+        elseif logFile(1).isStim==1
+            
+            namesExtraColumns = returnNamesExtraColumns(logFile);
+            isValid = ones(1,numel(namesExtraColumns))
+            for iExtraColumn = 1:numel(namesExtraColumns)
+                data = logFile(iEvent).(namesExtraColumns{iExtraColumn}); 
+                if isempty(data) || isnan(data) || ( ischar(data) && strcmp(data,'n/a') )
+                    isValid(iExtraColumn) = 0; 
+                end
+            end
+            if ~any(isValid) 
```

Comment:
```suggestion
            if all(~isValid) 
```

I find this a little easier to decipher when reading the code, closer to human language "if they are all not valid".